lib/connectors/mongodb/connector.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'mongo'
+require 'connectors/base/connector'
+
+module Connectors
+  module MongoDB
+    class Connector < Connectors::Base::Connector
+      SERVICE_TYPE = 'mongo'
+
+      def display_name
+        'MongoDB'
+      end
+
+      def configurable_fields
+        {
+           :host => {
+             :label => 'MongoDB Server Hostname'
+           },
+           :database => {
+             :label => 'MongoDB Database'
+           },
+           :collection => {
+             :label => 'MongoDB Collection'
+           }
+        }
+      end
+
+      def health_check(_params)
+        Connectors::MongoDB::CustomClient.new(hostname, database)
+      end
+
+      def sync(connector)
+        super(connector)
+
+        config = connector.configuration
+
+        host = config[:host][:value]
+        database = config[:database][:value]
+        collection = config[:collection][:value]
+
+        mongodb_client = create_client(host, database)
+
+        mongodb_client[collection].find.each do |document|
```

Comment:
This looks like it's bypassing the need for an Extractor. Is this a patter that we're going to be running with going forward?

Reply:
I disagree. We have a suite of connectors in Workplace Search that we're planning to move over here, right? We don't intend to re-write them, correct? So we will have the `Config`, `Adapter`, `Extractor` pattern to maintain. This is just making two models of code execution for us to maintain, and muddies the water on where logic should live.

> YAGNI

I wish we'd stop throwing this out all over the place. It's not a trump card. The Workplace Search Connectors show that we _are_ going to need this kind of abstraction as we grow. By throwing it away now, we're abandoning lessons-learned and assuming we can do better if we start from scratch. That's an anti-pattern.

Reply:
Just a bit of perspective on the Workplace Search connectors to add to the conversation

>  We have a suite of connectors in Workplace Search that we're planning to move over here, right? We don't intend to re-write them

Yes, but here are some subtle nuances to the yes. 

- The timeline to move them isn't set in stone, will our existing design patterns in Workplace Search be the ones we want to use if we port some over a year from now? 2? What could change in a longer timeline? We are likely going to be pushing for new connectors that branch out our use cases to be served.
- We likely won't port over all of them at the same time  because not all of them are equally worth the work to port over *cough* *cough* Zendesk.  I think our 8.2 release showed me that a lift and shift would have to be done in phases and we would need to challenge their structure to fit new challenges when we connect the dots to Elasticsearch. 
-  A rewrite is likely needed in some portion to be honest. Definitely we would need to rewrite for example Slack and Gmail as we won't be bringing forward concepts like private sources. 

I think the challenge we'll face is to balance being able to maximize how much of the useful connectors stuff we can bring over with the timeline to truly develop our new ingestion catalog identity. I don't expect that @danajuratoni 's project brief on the 8.5 selected connectors will or should request all of the Workplace Search ones, especially since we don't have Oauth support quite sorted yet, nor do we have indexing rules. If we're going to support native connectors, I consider Indexing rules very necessary as the right guardrails to give to customers who like to try and ingest ALL THE THINGs.

Reply:
Adding a short comment here too, since the retro session will happen in 2 weeks.
> We have a suite of connectors in Workplace Search that we're planning to move over here, right? We don't intend to re-write them, correct?

Along Serena's valid points - how much can be reused from the WS connectors and if the abstractions make sense is something that needs to be reviewed by the eng team, but I don't expect we'll be able to migrate them as they are. Not only because of the async model, but also since feature parity is not a goal (rather ease of use, configurability to name a few). e.g. we won't have 3 sync types as in WS.
I think everyone would benefit from close technical collaboration between working groups, so I'm looking forward to hearing suggestions in the retro!